### PR TITLE
feat(rpc): Add `address_` namespace and bindings

### PR DESCRIPTION
**Pull request by perama-v**
## Description
Progress toward:
- #4054 

## Changes
`address_` namespace addition including:
    - add `AddressApi` trait in rpc-api
    - add to RPC builder
    - add placeholder implementation for `address_getAppearances`

### CODE REVIEW

File: crates/rpc/rpc-api/src/address.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+use jsonrpsee::{core::RpcResult, proc_macros::rpc};
+use reth_primitives::{Address, BlockNumber, TxIndex};
+
+/// address_ namespace RPC interface.
+#[cfg_attr(not(feature = "client"), rpc(server, namespace = "address"))]
+#[cfg_attr(feature = "client", rpc(server, client, namespace = "address"))]
+pub trait AddressApi {
+    /// Returns transaction identifiers relevant to an address.
+    ///
+    /// See: https://github.com/ethereum/execution-apis/pull/453/
```
Comment (mattsse) [inline crates/rpc/rpc-api/src/address.rs:10]:
```suggestion
    /// See: <https://github.com/ethereum/execution-apis/pull/453/>
```

File: crates/rpc/rpc-api/src/address.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+    #[method(name = "getAppearances")]
+    async fn get_appearances(
+        &self,
+        address: Address,
+        range: Option<(BlockNumber, BlockNumber)>,
+    ) -> RpcResult<Vec<(BlockNumber, TxIndex)>>;
```
Comment (mattsse) [inline crates/rpc/rpc-api/src/address.rs:16]: there's no spec for this yet, so we should think about some examples here and how this is parsed.

should the range be an object?
and the response a struct with named fields?

**Review by mattsse — CHANGES_REQUESTED** [review]
supportive, since this doesn't add any overhead,

but like to see some serde examples how the request response objects should look like

**Review by onbjerg — CHANGES_REQUESTED** [review]
Would want this to be documented fully in the book (e.g. in `book/jsonrpc`) in the same PR since it isn't a standard namespace

File: crates/rpc/rpc-api/src/address.rs (hunk 2 above)
Comment (perama-v) [reply crates/rpc/rpc-api/src/address.rs]: Good suggestion. I have changed the range to be an object. 

The return type is also now spec compliant (lives in ethereum/execution-apis). Serde unit tests added.

**Comment by perama-v** [comment]
> Would want this to be documented fully in the book (e.g. in `book/jsonrpc`) in the same PR since it isn't a standard namespace

Docs added and examples included that mirror the spec/spectests

**Comment by gakonst** [comment]
Just to make this explicit would like to be sensitive with merging this upstream and maintaining the code..so I am happy we're iterating on it and would like to ideally figure out how to run it as a separate binary as well?

**Comment by perama-v** [comment]
> Just to make this explicit would like to be sensitive with merging this upstream and maintaining the code..so I am happy we're iterating on it and would like to ideally figure out how to run it as a separate binary as well?

Thanks, appreciate that. 

I consider this family (4054) an exploration of "what could this feature look like, and how could it be implemented?", with the outcome being discovery of whether it is a good feature and what a good implementation could look like. 

Happy to explore the design space including specific ideas anyone has.

Could you expand the concept of a separate binary more?  On first take it sounds like you are describing the following:

Currently, the Execution stage has inspectors that can gather data. The result of the inspector could be stored directly in to a table alongside other tables in reth, alternatively it could call a separate binary and pass the inspector data over. The separate binary listens for and accepts the data, processes and stores in some representation that reth is unaware of. Then to serve the method reth and the separate binary again call each other (depending on where the call is set up to be served from) and the data aggregated for the response.

**Comment by gakonst** [comment]
Closing this as stale, we'll re-incorporate these ideas as we venture into custom stages with Reth CLI Extensions and the Declarative Node Builder API that @mattsse is spearheading.

**Pull request closed without merging**